common_util/src/partitioned_lock.rs

```diff
@@ -7,6 +7,7 @@ use std::{
     sync::{Mutex, MutexGuard, RwLock, RwLockReadGuard, RwLockWriteGuard},
 };
 
+/// Warning, do not use this hash in non-memory scenarios.
```

Comment:
This warning should be place in where AHasher is defined, not its usage.

Also you can link to its readme to explain its reason.
https://github.com/tkaitchuck/aHash/blob/master/README.md#goals-and-non-goals